### PR TITLE
some stock optimizations

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph 
cc @twymer @czue

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -39,7 +39,7 @@ pytz==2014.10
 PyYAML==3.10
 raven==3.1.17
 requests==2.0.0
-south==0.7.3
+south==1.0.2
```
Comment (snopoke) [inline requirements/requirements.txt]: I tested this on staging and was able to deploy without any issues. Also reviewed the change log and didn't see anything that should break (so did @emord: https://github.com/dimagi/commcare-hq/pull/6794/files#r30916913)

**Comment by czue** [comment]
:+1: from me once tests pass (didn't review super closely though)

**Comment by snopoke** [comment]
Updated to remove the new index (will make a separate PR). This should be gtg.

File: corehq/ex-submodules/casexml/apps/stock/utils.py [hunk 2]
```diff
@@ -45,19 +45,33 @@ def state_stock_category(state):
...
+    }
     """
-    trans = get_current_ledger_transactions_multi([case_id])
-    return trans[case_id]
+    from corehq.apps.commtrack.models import StockState
+    results = StockState.objects.filter(case_id=case_id).values_list('case_id', 'section_id', 'product_id')
+
+    ret = {}
+    for case_id, section_id, product_id in results:
+        sections = ret.setdefault(section_id, {})
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/utils.py]: just for my own education, any reason to use this over `defaultdict`?

File: corehq/ex-submodules/casexml/apps/stock/utils.py [hunk 3]
```diff
@@ -45,19 +45,33 @@ def state_stock_category(state):
 def get_current_ledger_transactions(case_id):
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/stock/utils.py]: my guess is that we may be able to remove this method entirely and update the callers to use states too but that's definitely out of scope

**Comment by czue** [comment]
good to merge once tests pass

File: corehq/ex-submodules/casexml/apps/stock/utils.py (hunk 2 above)
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/stock/utils.py]: No reason in this case.

File: corehq/ex-submodules/casexml/apps/stock/utils.py (hunk 3 above)
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/stock/utils.py]: Yea, makes sense. I think this is the only place that actually needs data which isn't in the `StockState` (it uses `transaction.report.form_id` in the template): https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/case/templatetags/case_tags.py#L95

**Comment by czue** [comment]
kicked build

**Pull request merged**